tscat_gui/edit.py

```diff
@@ -0,0 +1,70 @@
+from PySide2 import QtCore, QtWidgets
+
+from .utils.keyword_list import EditableKeywordListWidget
+
+import tscat
+import datetime as dt
+
+
+class AttributeGroupBox(QtWidgets.QGroupBox):
+    def __init__(self,
+                 title: str,
+                 attributes: list[str],
+                 catalogue: tscat.Catalogue,
+                 parent: QtWidgets.QWidget = None):
+        super().__init__(title, parent)
+
+        self.catalogue = catalogue
+
+        layout = QtWidgets.QGridLayout()
+        layout.setMargin(0)
+
+        for row, attr in enumerate(attributes):
+            layout.addWidget(QtWidgets.QLabel(attr.title()), row, 0)
+            value = catalogue.__dict__[attr]
+            if type(value) == str:
```

Comment:
I might prefer something with a dict as lookup table with type->widget ctor as key/value.
That would allow to share this map of std widgets for each type.

then you could have something like this:
```python
# in some dedicated module:
def float_delegate(value):
    sb = QtWidgets.QDoubleSpinBox()
    sb.setValue(value)
    sb.setRange(float('-inf'), float('inf'))
    return sb

str_delegate= QtWidgets.QLineEdit
    
default_delegate = lambda value: QtWidgets.QLabel(f'TODO {type(value)} {value}')

ctor_table = {
    float: float_delegate,
    str: str_delegate,
}


# then here:

def append_delegate(args):
    row, attribute = args
    layout.addWidget(QtWidgets.QLabel(attribute.title()), row, 0)
    value = catalogue.__dict__[attribute]
    delegate_ctor = ctor_table.get(type(value), default_delegate)
    layout.addWidget(delegate_ctor(value), row, 1)

list(map(append_delegate, enumerate(attributes)))
# or a simple loop if you prefer
```

Reply:
Good idea. Should we name it factory? Delegate-factory. Ctor has another meaning.